backend/repository/endpoint_update.go

```diff
@@ -69,6 +69,114 @@ func (r *Repository) WriteEndpointConfig(
 	return nil
 }
 
+// DeleteEndpointConfig deletes an individual endpoint config, updates endpoint group config and endpoint meta json
+// endpointName must have the form endpointId.handleId
+func (r *Repository) DeleteEndpointConfig(endpointCfgDir string, endpointName string) error {
+	gatewayCfg, err := r.LatestGatewayConfig()
+	if err != nil {
+		return errors.Wrap(err, "loading gateway config error")
+	}
+	nameParts := strings.Split(endpointName, ".")
```

Comment:
dependency resolution should be in one uniq pkg, answer such question of what depend on what, use case I can think of:

- partial tests
- delete endpoints
- notify client owners for new user case 

... and more

it should not be each module implement their own logic like this

`codegen.ModuleDep` pkg, expose functions 
`GetDepFrom(client, endpoint)` or something

@ChuntaoLu